sphinx/source/docs/dev_guide/server.rst

```diff
@@ -258,8 +259,8 @@ two sequences of frames, one sequence in each direction ("full duplex").
 On top of websocket frames, we implement our own ``Message`` concept. A Bokeh
 ``Message`` spans multiple websocket frames. It always contains a header frame,
 metadata frame, and content frame. These three frames each contain a JSON
-string. The code permits these three frames to be followed by binary data
-frames, but currently in Bokeh binary data frames are not used.
+string. The code permits these three frames to be followed by optional binary data
+frames. Currently in Bokeh, binary data frames are only used to transport image data.
```

Comment:
Also, to be technically pedantic, the original statement is still correct. @philippjfrallowed for a different encoding of numpy arrays, but array data is still sent as part of the message "content" block, as an encoded string. It just happens to be much faster to encode/decode the string than raw JSON encoding of the array was. 

Separate from this, the Bokeh protocol also allows for multi-frame messages. In principle this could allow for example, for sending numpy arrays directly from their memory buffers to the websocket with no additional copies. This might offer even more performance boost. But it is not used yet. There is nothing in Bokeh that makes use of binary frames yet.